PAY-5859: Payment Services drop-in documentation for Summit (#170)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/payment-services/containers/credit-card.mdx

---
title: CreditCard container

> description: Configure the CreditCard container to manage and display the available credit card fields during a checkout.

Reviewer: > Configure the CreditCard container to manage and display the available [...] fields [...]

When reading this, I get the impression that the `CreditCard` container renders those fields that are "available". This seems confusing as there is no such thing as an "available field". Instead, it's all or nothing: either the credit card _form_ is available (with its three inputs: number+expdate+cvv), or it is not.

```suggestion
description: Use the CreditCard container to render a credit card form during checkout.
```

---

import OptionsTable from '@components/OptionsTable.astro';

> The `CreditCard` container allows a shopper to place an order with credit card details during the checkout process. You can configure it to manage and display the available credit card fields for a payment method.

Reviewer: "available credit card fields"

Same. There is no such thing as an "available" credit card field.

---
title: CreditCard container
description: Use the CreditCard container to render a credit card form during checkout.
---

import OptionsTable from '@components/OptionsTable.astro';

> The `CreditCard` container allows a shopper to place an order with credit card details during the checkout process. You can configure it to manage and display the credit card fields.

Reviewer 2: ```suggestion
The `CreditCard` container allows a shopper to enter credit card details during the checkout process. You can configure it to manage and display the credit card fields.
```

## CreditCard configurations

The `CreditCard` container provides the following configuration options:

<OptionsTable
  compact
  options={[
    ['Option', 'Type', 'Req?', 'Description'],
    ['apiUrl', 'string', 'Yes', 'The URL to the Adobe Commerce GraphQL endpoint, such as "https://example.com/graphql".'],
    ['getCartId', 'function', 'Yes', 'Should return a promise that resolves to the shopper`s cart ID.'],

>     ['getCustomerToken', 'function', 'Yes', 'The credit card container may send GraphQL requests on behalf of the shopper. This requires GraphQL authorization, which can be performed using authorization tokens or session cookies. For token-based authorization, the "getCustomerToken" function should return a customer token as a string or null for guest checkouts. For session-based authorization, the "getCustomerToken" property must explicitly be set to "null".'],

Reviewer 2: ```suggestion
    ['getCustomerToken', 'function', 'Yes', 'The credit card container may send GraphQL requests on behalf of the shopper. This requires GraphQL authorization, which can be performed using authorization tokens or session cookies. For token-based authorization, the "getCustomerToken" function should return a customer token as a string, or null for guest checkouts. For session-based authorization, the "getCustomerToken" property must explicitly be set to "null".'],
```

    ['creditCardFormRef', 'object', 'Yes', 'Credit card form reference. Initially, { current: null } should be passed. Once rendered, the credit card container will set the "current" property to an { validate: () => boolean; submit: () => Promise<void> } object, which parent containers can use to (programmatically) validate and submit the credit card form.'],

>     ['onSuccess', 'function', 'No', 'Called when payment flow finishes successfully. This function is not passed any arguments.'],

Reviewer 2: ```suggestion
    ['onSuccess', 'function', 'No', 'Called when the payment flow finishes successfully. This function is not passed any arguments.'],
```
